### PR TITLE
creating a new shopping list

### PR DESCRIPTION
**Pull request by didemydn**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

A shopping list is a set of items associated with the owner's unique id and the name of the shopping list. Users need to be able to create new shopping lists. To do this, the `Home` view should present them with a form that allows them to enter the name of their list. Additionally, we need to do a few things to create that list:

## Related Issue

issue #4 

## Acceptance Criteria

UI-related tasks:
- [x] The `Home` view displays a form that allows users to enter the name of a shopping list and then create a list with that name.
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x]  When the user submits the form, they see a message indicating that the list either was or was not created and saved to the database.

Data-related tasks:
- [x] Clicking the button uses the `createList` function in `src/api/firebase.js` to create a new shopping list for the user.
- [x] The shopping list path is stored in local storage using the `setListPath` function.
- [x] Once the list has been created and saved, the user is redirected to the `List` view.

### Before

Home view, there is no list label and create a list button. 
image

### After
creating list button and form 
image

after writing a list name you should navigate to list view
image

Home view after creating a list 
image

## Testing Steps / QA Criteria

- npm start 
- fill the form to create a list on the home view

### CODE REVIEW

**Review by GrannyYetta — COMMENTED** [review]
The feature works as expected, thank you for the comments in the code. 

We had the idea that later on we can implement toast notifications to display the messages. Given the redirection to the /list view, the messages are not visibly displayed.

Looking forward to the code review!

**Comment by eva-lng** [comment]
The feature for creating new lists works well, code is looks good and is easy to understand. As GrannyYetta mentionted, in the future we could implement toaster for all notifications. Right now the user doesn't see any success message because they are redirected to the List view right after submitting the form.

**Review by eva-lng — APPROVED** [review]
The feature for creating new lists works well, code is looks good and is easy to understand. As GrannyYetta mentionted in the comments, in the future we could implement toaster for all notifications. Right now the user doesn't see any success message because they are redirected to the List view right after submitting the form.

File: src/views/Home.jsx [hunk 1]
```diff
@@ -1,7 +1,35 @@
...
+
+			const createListPath = `${userId}/${listName}}`;
+			setListPath(createListPath);
+			navigate('/list'); //redirect to the list view
+		} catch (error) {
+			//Logging and error messages if list is not created
+			console.error('error creating a list', error);
+			setMessage('Failed to create list. Please try again!');
+		}
+	};
```
Comment (Amaka202) [inline src/views/Home.jsx:31]: Good job on this function team! it is a pleasant way to handle async calls!

**Review by Amaka202 — APPROVED** [review]
Nice job team. Your feature works well. 💥💥

Something to consider if you have the time now or later during styling/refactoring (last 2 weeks) is adding a timer to show the success or error message before redirecting the users to the list page. Cause now I dont get to see the success message after adding, it just takes me to list page. But if a timer is in place I will see it before it redirects.

The timer will also help for the error message too cause now if I get an error it never goes away whereas if there was a timer the error message will show for like 5 seconds and then clear by itself.

**Pull request merged**